sdk_v2/kubeflow/training/api/training_client.py

```diff
@@ -12,7 +12,540 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+import multiprocessing
+import queue
+import random
+import string
+import uuid
+from typing import Callable, Dict, List, Optional
+
+from kubeflow.training import models
+from kubeflow.training.api_client import ApiClient
+from kubeflow.training.constants import constants
+from kubeflow.training.types import types
+from kubeflow.training.utils import utils
+from kubernetes import client, config, watch
+
+logger = logging.getLogger(__name__)
+
 
 class TrainingClient:
-    def __init__(self):
-        print("Work in progress...")
+    def __init__(
+        self,
+        config_file: Optional[str] = None,
+        context: Optional[str] = None,
+        client_configuration: Optional[client.Configuration] = None,
+        namespace: str = utils.get_default_target_namespace(),
+    ):
+        """TrainingClient constructor. Configure logging in your application
+            as follows to see detailed information from the TrainingClient APIs:
+            .. code-block:: python
+                import logging
+                logging.basicConfig()
+                log = logging.getLogger("kubeflow.training.api.training_client")
+                log.setLevel(logging.DEBUG)
+
+        Args:
+            config_file: Path to the kube-config file. Defaults to ~/.kube/config.
+            context: Set the active context. Defaults to current_context from the kube-config.
+            client_configuration: Client configuration for cluster authentication.
+                You have to provide valid configuration with Bearer token or
+                with username and password. You can find an example here:
+                https://github.com/kubernetes-client/python/blob/67f9c7a97081b4526470cad53576bc3b71fa6fcc/examples/remote_cluster.py#L31
+            namespace: Target Kubernetes namespace. If SDK runs outside of Kubernetes cluster it
+                takes the namespace from the kube-config context. If SDK runs inside
+                the Kubernetes cluster it takes namespace from the
+                `/var/run/secrets/kubernetes.io/serviceaccount/namespace` file. By default it
+                uses the `default` namespace.
+        """
+
+        # If client configuration is not set, use kube-config to access Kubernetes APIs.
+        if client_configuration is None:
+            # Load kube-config or in-cluster config.
+            if config_file or not utils.is_running_in_k8s():
+                config.load_kube_config(config_file=config_file, context=context)
+            else:
+                config.load_incluster_config()
+
+        k8s_client = client.ApiClient(client_configuration)
+        self.custom_api = client.CustomObjectsApi(k8s_client)
```

Comment:
Does these client include retries on any failure? May be we could have a wrapper client that would add retries

Reply:
Yes, I think it is a good idea.
@akshaychitneni Can we have a followup PR to implement it ?
Please can you create an issue to track it ?